### PR TITLE
cmake: update minimum boost version to 1.66

### PR DESCRIPTION
**Pull request by cbodley**
fixes compilation failures in radosgw with boost 1.66, and pins that as the required version so we don't have to continue supporting beast as a submodule

Fixes: http://tracker.ceph.com/issues/22600

### CODE REVIEW

**Comment by cbodley** [comment]
this contains the cmake changes for boost 1.66 from https://github.com/ceph/ceph/pull/19054, which were reviewed and tested there

File: src/include/config-h.in.cmake [hunk 1]
```diff
@@ -330,4 +330,7 @@
 /* Defined if getentropy() is available */
 #cmakedefine HAVE_GETENTROPY
 
+/* Defined if boost::context is available */
+#cmakedefine HAVE_BOOST_CONTEXT
```
Comment (tchaikov) [inline src/include/config-h.in.cmake]: @cbodley is this macro used anywhere? or i am just missing something.

Comment (tchaikov) [reply src/include/config-h.in.cmake]: sorry, nvm. i found it at https://github.com/ceph/ceph/pull/19054/files#diff-b5fd6691c22ab2c09a283b8bdda0e9ecR22

**Comment by wjwithagen** [comment]
@cbodley 
Got this working under FreeBSD... 
To replicate this is a bit harder

Mainly due to the fact that Boost 1.66 has not made it in the current set of packages. So I've build my own packages for Boost and matching Cmake.  But it doable.

**Pull request merged**